streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -310,53 +248,53 @@ void shutdown(final boolean clean) {
     }
 
     public Set<TaskId> previousRunningTaskIds() {
-        return active.previousRunningTaskIds();
+        return tasks.values()
+                    .stream()
+                    .filter(t -> t.isActive() && t.state() == Task.State.SUSPENDED)
+                    .map(Task::id)
+                    .collect(Collectors.toSet());
     }
 
     public Set<TaskId> activeTaskIds() {
-        return active.allAssignedTaskIds();
+        return tasks.values()
+                    .stream()
+                    .filter(Task::isActive)
+                    .map(Task::id)
+                    .collect(Collectors.toSet());
     }
 
     Set<TaskId> standbyTaskIds() {
-        return standby.allAssignedTaskIds();
-    }
-
-    Set<TaskId> revokedActiveTaskIds() {
-        return revokedActiveTasks.keySet();
-    }
-
-    Set<TaskId> revokedStandbyTaskIds() {
-        return revokedStandbyTasks.keySet();
-    }
-
-    Set<TaskId> previousActiveTaskIds() {
-        final HashSet<TaskId> previousActiveTasks = new HashSet<>(assignedActiveTasks.keySet());
-        previousActiveTasks.addAll(revokedActiveTasks.keySet());
-        previousActiveTasks.removeAll(addedActiveTasks.keySet());
-        return previousActiveTasks;
-    }
-
-    Set<TaskId> previousStandbyTaskIds() {
-        final HashSet<TaskId> previousStandbyTasks = new HashSet<>(assignedStandbyTasks.keySet());
-        previousStandbyTasks.addAll(revokedStandbyTasks.keySet());
-        previousStandbyTasks.removeAll(addedStandbyTasks.keySet());
-        return previousStandbyTasks;
+        return tasks.values()
+                    .stream()
+                    .filter(t -> t instanceof StandbyTask)
+                    .map(Task::id)
+                    .collect(Collectors.toSet());
     }
 
-    StreamTask activeTask(final TopicPartition partition) {
-        return active.runningTaskFor(partition);
+    StreamTask streamTask(final TopicPartition partition) {
+        for (final Task task : tasks.values()) {
+            if (task instanceof StreamTask && task.partitions().contains(partition)) {
+                return (StreamTask) task;
+            }
+        }
+        return null;
     }
 
     StandbyTask standbyTask(final TopicPartition partition) {
-        return standby.runningTaskFor(partition);
+        for (final Task task : tasks.values()) {
+            if (task instanceof StandbyTask && task.partitions().contains(partition)) {
+                return (StandbyTask) task;
+            }
+        }
+        return null;
     }
 
-    Map<TaskId, StreamTask> activeTasks() {
-        return active.runningTaskMap();
+    Map<TaskId, StandbyTask> standbyTasks() {
```

Comment:
Here I've consolidated the activeTasks / allStreamsTasks by filtering on `instanceOf` only.

Reply:
FYI, I changed this up a little to avoid repeated iterations.